### DISCUSSION TITLE
Allow compsets with stub atmosphere

### DISCUSSION THREAD

**Discussion by xylar** [E3SM-Project/E3SM#5794] [Development discussion]
I created a compset that is an MPAS-Ocean C-case with a stub atmosphere:
https://github.com/xylar/E3SM/commit/693c4778a5f3f3c0719162f05062c2e662ba1375

```
  <compset>
    <alias>CMPASO-S-ISMF</alias>
    <lname>2000_SATM_SLND_SICE_MPASO%ISMFDATMFORCED_SROF_SGLC_SWAV</lname>
    <support_level>"Experimental, under development"</support_level>
  </compset>
```

Such a configuration would be a very useful starting point for idealized ocean testing within E3SM.  Similar configurations would be useful for idealized sea ice-ocean and ice sheet-ocean testing.

However, this configuration currently results in an error:
```
ERROR:
  3:  (seq_mct_drv) ERROR: if prognostic surface model must also have atm present
```

https://github.com/E3SM-Project/E3SM/blob/88ee1863a2ca44bc142809a43ea427cfa381d567/driver-mct/main/cime_comp_mod.F90#L1895

(I have investigated the `CPL_BYPASS` CPP flag around this line and that doesn't seem like a plausible route to fixing this particular issue.)

I am hoping to get some help understanding why this check is there and what the unforeseen consequences might be of removing it.

**Comment by xylar** [comment]
@rljacob, you mentioned at the All-hands Meeting that you thought a stub atmosphere was something we could support. I'm hoping to discuss more.

**Comment by xylar** [comment]
If I attempt to bypass this line, I next see:
```
 64: MCT::m_AttrVect::Copy_MCTERROR: Input aV and output aV do not have the same size
 64: MCT::m_AttrVect::Copy_: MCTERROR: Input aV and output aV do not have the same size error, stat =2
```
which I have traced to here in MCT:
https://github.com/MCSclimate/MCT/blob/e36024c5ddf482625ae6bd9474eff7d8f393f87c/mct/m_AttrVect.F90#L2542-L2547
but which doesn't provide a helpful traceback.

**Reply by rljacob** [reply]
Found this thread while looking at the discussions section. What was the grid for this case?  I'll try it and find out more about this error.

**Comment by rljacob** [comment]
That first error message is because the coupler assumes any surface model would gets its forcing from the atmosphere and a prognostic surface model would be a "real" global model, not an idealized config.  How would this case get its forcing?

**Reply by xylar** [reply]
3D Restoring of ocean properties to a prescribed data field and fluxes at the ocean surface under ice shelves.

**Comment by erinethomas** [comment]
@xylar  -  did you ever solve your MCT error? I just ran into this for the wave coupling and have a fix if you need it.

**Reply by xylar** [reply]
@erinethomas , no I gave up and would appreciate hearing about your fix!

**Comment by erinethomas** [comment]
Like you, I got zero useable traceback information for debugging. I added some "divide by zero" lines of code right above the 'die'
 commands in m_AttrVect.F90 (there are five places where it checks if Input aV and output aV have the same size or not) that forced the model to crash with useful traceback info. 

In my case, it was because I had added variables for wave coupling but was testing a configuration with No Wave component (i.e the input array was not the same as the output array)
